src/ethereum/state_oracle/interface.py

```diff
@@ -0,0 +1,213 @@
+"""
+Abstract interface for state oracle implementations.
+"""
+
+from typing import Any, Callable, Dict, Optional, Protocol, Tuple
+
+from ethereum_types.bytes import Bytes20, Bytes32
+
+# Use generic types for compatibility across forks
+Account = Any
+Address = Bytes20
```

Comment:
I initially thought that this interface would fork agnostic, but perhaps it makes sense to have one per fork?